File: VideoHub/backend/internal/model/media.go
package model

import (
	"time"
)

type MediaType string

const (
	MediaTypeMovie MediaType = "movie"
	MediaTypeTV    MediaType = "tv"
	MediaTypeAnime MediaType = "anime"
)

type MediaStatus string

const (
	MediaStatusPending   MediaStatus = "pending"
	MediaStatusScraped   MediaStatus = "scraped"
	MediaStatusFailed    MediaStatus = "failed"
)

type Media struct {
	ID            int64       `gorm:"primaryKey;autoIncrement" json:"id"`
	Type          MediaType   `gorm:"type:text;not null;index" json:"type"`
	Title         string      `gorm:"type:text;not null" json:"title"`
	OriginalTitle string      `gorm:"type:text" json:"original_title,omitempty"`
	Year          int         `gorm:"type:integer;index" json:"year,omitempty"`
	Path          string      `gorm:"type:text;not null;uniqueIndex" json:"path"`
	PosterPath    string      `gorm:"type:text" json:"poster_path,omitempty"`
	BackdropPath  string      `gorm:"type:text" json:"backdrop_path,omitempty"`
	Overview      string      `gorm:"type:text" json:"overview,omitempty"`
	Rating        float64     `gorm:"type:real;default:0;index" json:"rating,omitempty"`
	Runtime       int         `gorm:"type:integer;default:0" json:"runtime,omitempty"`
	Genres        string      `gorm:"type:text" json:"genres,omitempty"`
	TmdbID        int64       `gorm:"type:integer;index" json:"tmdb_id,omitempty"`
	ImdbID        string      `gorm:"type:text" json:"imdb_id,omitempty"`
	Status        MediaStatus `gorm:"type:text;default:'pending';index" json:"status"`
	Seasons       []Season    `gorm:"foreignKey:SeriesID" json:"seasons,omitempty"`
	CreatedAt     time.Time   `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt     time.Time   `gorm:"autoUpdateTime" json:"updated_at"`
}

type Season struct {
	ID           int64      `gorm:"primaryKey;autoIncrement" json:"id"`
	SeriesID     int64      `gorm:"not null;index" json:"series_id"`
	SeasonNumber int        `gorm:"not null" json:"season_number"`
	Title        string     `gorm:"type:text" json:"title,omitempty"`
	PosterPath   string     `gorm:"type:text" json:"poster_path,omitempty"`
	Episodes     []Episode  `gorm:"foreignKey:SeriesID,SeasonNumber" json:"episodes,omitempty"`
	CreatedAt    time.Time  `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt    time.Time  `gorm:"autoUpdateTime" json:"updated_at"`
}

type Episode struct {
	ID            int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	SeriesID      int64     `gorm:"not null;index:idx_series_id" json:"series_id"`
	SeasonNumber  int       `gorm:"not null" json:"season_number"`
	EpisodeNumber int       `gorm:"not null" json:"episode_number"`
	Title         string    `gorm:"type:text" json:"title,omitempty"`
	Path          string    `gorm:"type:text;not null;uniqueIndex" json:"path"`
	PosterPath    string    `gorm:"type:text" json:"poster_path,omitempty"`
	Overview      string    `gorm:"type:text" json:"overview,omitempty"`
	Runtime       int       `gorm:"type:integer;default:0" json:"runtime,omitempty"`
	TmdbID        int64     `gorm:"type:integer" json:"tmdb_id,omitempty"`
	AirDate       string    `gorm:"type:text" json:"air_date,omitempty"`
	CreatedAt     time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt     time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

File: VideoHub/frontend/src/types/media.ts
export interface Media {
  id: number
  type: 'movie' | 'tv' | 'anime'
  title: string
  original_title?: string
  year?: number
  path: string
  poster_path?: string
  backdrop_path?: string
  overview?: string
  rating?: number
  runtime?: number
  created_at?: string
  updated_at?: string
}

export interface Episode {
  id: number
  series_id: number
  season_number: number
  episode_number: number
  title?: string
  path: string
  poster_path?: string
  overview?: string
  runtime?: number
}

export interface WatchHistory {
  id: number
  media_id: number
  episode_id?: number
  progress: number
  completed: boolean
  last_watched: string
  media?: Media
  episode?: Episode
}

export interface Favorite {
  id: number
  media_id: number
  created_at: string
  media?: Media
}

export interface MediaListParams {
  type?: 'movie' | 'tv' | 'anime'
  sort?: 'title' | 'year' | 'rating' | 'created_at'
  order?: 'asc' | 'desc'
  page?: number
  limit?: number
  search?: string
}

export interface Subtitle {
  id: number
  name: string
  url: string
}

export interface Folder {
  id: number
  name: string
  path: string
}

export interface PlayUrl {
  url: string
}